### DISCUSSION TITLE
What Release Pattern Should Be Used to Build Images?

### DISCUSSION THREAD

**Discussion by v1ferrarij** [actions/virtual-environments#7199] [General questions]
We are looking to build Self Hosted Agents using this repository.

Should we only build images when a new release branch/tag is made for our relevant OS, or should we build from main on a weekly basis for example?

**Comment by panekj** [comment]
Generally you should always build the latest commit and never use specific tag, that's because `HEAD` contains latest fixes to build process; that can fail for various reasons; as well as latest software version.
As for the frequency it's up to you, but generally if build fails, always try to grab latest commit first.
https://github.com/actions/runner-images/discussions/7089
